rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -158,37 +158,49 @@ private RepositorySnapshot getCurrentRepository() {
         return repositoryLocator.snapshotAt(getBestBlock().getHeader());
     }
 
-    @Override
-    public synchronized List<Transaction> addTransactions(final List<Transaction> txs) {
+    private List<Transaction> addSuccesors(Transaction tx) {
         List<Transaction> added = new ArrayList<>();
+        Optional<Transaction> succesor = this.getQueuedSuccesor(tx);
 
-        for (Transaction tx : txs) {
-            if (this.addTransaction(tx).transactionWasAdded()) {
-                added.add(tx);
-
-                Optional<Transaction> succesor = this.getQueuedSuccesor(tx);
-
-                while (succesor.isPresent()) {
-                    Transaction found = succesor.get();
-                    queuedTransactions.removeTransactionByHash(found.getHash());
+        while (succesor.isPresent()) {
+            Transaction found = succesor.get();
+            queuedTransactions.removeTransactionByHash(found.getHash());
 
-                    if (!this.addTransaction(found).transactionWasAdded()) {
-                        break;
-                    }
+            if (!this.internalAddTransaction(found).transactionWasAdded()) {
+                break;
```

Comment:
should we `break` here? `getQueuedSuccesor` method returns the first tx from the collection of transactions with the same nonce. There might be some other tx with the same nonce which potentially could be added? Another question: should the `getQueuedSuccesor` method return the first tx or the oldest one (from the list of txs with the same nonce)?

Reply:
re: _There might be some other tx with the same nonce which potentially could be added?_, NO, but because the queuedTransactions collection is a TransactionSet, which is a particular class that doesn't allow more than one tx with the same nonce, see: https://github.com/rsksmart/rskj/blob/918e14623851fc58aaf3b7a39d2ef680adbc686f/rskj-core/src/main/java/org/ethereum/core/TransactionSet.java#L44

The way the method works is that the new tx always removes any preexisting tx with the same nonce in the TransactionSet.

Reply:
Originally, transaction pool allowed many transactions from the same sender with the same nonce (with different values, and then, different hash). The last implementors decided to not allow such case. I'm not sure it is the right way. Maybe the client/user changed their intention, and the put another competing transaction, to replace an old one. Maybe we should review other implementations (geth, parity). 

But now, there are no transactions with same sender, same nonce